pkg/util/trace/report_statement.go

```diff
@@ -81,10 +99,74 @@ func (s StatementInfo) CsvFields() []string {
 	result = append(result, GetNodeResource().NodeUuid)
 	result = append(result, GetNodeResource().NodeType)
 	result = append(result, nanoSec2DatetimeString(s.RequestAt))
-	result = append(result, s.ExecPlan)
+	result = append(result, nanoSec2DatetimeString(s.ResponseAt))
+	result = append(result, fmt.Sprintf("%d", s.Duration))
+	result = append(result, s.Status.String())
+	if s.Error == nil {
+		result = append(result, "")
+	} else {
+		result = append(result, fmt.Sprintf("%s", s.Error))
+	}
+	result = append(result, s.ExecPlan2Json())
+
 	return result
 }
 
+func (s *StatementInfo) ExecPlan2Json() string {
+	if s.ExecPlan == nil {
+		return "{}"
+	}
+	json, err := json.Marshal(s.ExecPlan)
+	if err != nil {
+		return fmt.Sprintf(`{"err": %q}`, err.Error())
+	}
+	return string(json)
+}
+
+// SetExecPlan record execPlan should be TxnComputationWrapper.plan obj, which support 2json.
+func (s *StatementInfo) SetExecPlan(execPlan any) {
+	s.ExecPlan = execPlan
+}
+
+func (s *StatementInfo) SetTxnIDIsZero(id []byte) {
+	if bytes.Equal(s.TransactionID[:], nilTxnID[:]) {
+		copy(s.TransactionID[:], id)
+	}
+}
+
+func (s *StatementInfo) Report(ctx context.Context) {
+	s.reported = true
+	ReportStatement(ctx, s)
+}
+
+var EndStatement = func(ctx context.Context, err error) time.Time {
+	s := StatementFromContext(ctx)
+	if s == nil {
+		panic(moerr.NewInternalError("no statement info in context"))
+	}
+	endTime := util.NowNS()
+	if s.end {
+		goto endL
```

Comment:
I like goto, but here, isn't that you put all the // do report stuff under if !s.end { }, then no need of goto, right?